src/core/ext/filters/client_channel/xds/xds_api.cc

```diff
@@ -304,9 +314,14 @@ grpc_slice XdsApi::CreateLdsRequest(const std::string& server_name,
                                     bool populate_node) {
   upb::Arena arena;
   envoy_api_v2_DiscoveryRequest* request =
-      CreateDiscoveryRequest(arena.ptr(), kLdsTypeUrl, version, nonce, error,
-                             populate_node ? node_ : nullptr,
-                             populate_node ? build_version_ : nullptr);
+      CreateDiscoveryRequest(arena.ptr(), kLdsTypeUrl, version, nonce, error);
+  // Populate node.
+  if (populate_node) {
+    envoy_api_v2_core_Node* node_msg =
```

Comment:
suggesting passing request into PopulateNode and move 
envoy_api_v2_core_Node* node_msg =
        envoy_api_v2_DiscoveryRequest_mutable_node(request, arena.ptr());
into PopulateNode in all 4 places

Reply:
I thought about that, but it won't work, because in `XdsApi::CreateLrsInitialRequest()`, the outer message is a `LoadStatsRequest` instead of a `DiscoveryRequest`.